ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-text-input/activityTextInput.component.ts

```diff
@@ -89,11 +89,11 @@ export class ActivityTextInput implements OnInit, OnChanges, OnDestroy {
         const adjustedOffset = tagsAddedLength + match.offset;
 
         taggedValue =
-          taggedValue.substr(0, adjustedOffset) +
+          taggedValue.substring(0, adjustedOffset) +
           openTag +
-          taggedValue.substr(adjustedOffset, match.length) +
+          taggedValue.substring(adjustedOffset, adjustedOffset + match.length) +
```

Comment:
I think this might be wrong. Please check.

Reply:
Alright. 
The `substr()` method begins at a specified position, and returns a specified number of characters, e.g. `substr(startIndex, length)`.
The `substring()` method extracts characters from start to end (endIndex is not included), e.g. `substring(startIndex, endIndex)`.
Both methods don't change the original string.

case 1. Cut from the string beginning:
```
'abcdefg'.substr(0, 3) -> 'abc'
'abcdefg'.substring(0, 0+3) -> 'abc'
```

case 2. Cut from the string middle:
```
'abcdefg'.substr(2, 3) -> 'cde'
'abcdefg'.substring(2, 2+3) -> 'cde'
```

case 3. Cut from the string middle to the string end (second argument is undefined):
```
'abcdefg'.substr(4) -> 'efg'
'abcdefg'.substring(4) -> 'efg'
```

case 4. Cut from the string middle to the string end (second argument is bigger then endIndex/string length):
```
'abcdefg'.substr(3, 58) -> defg'
'abcdefg'.substring(3, 3+58) -> 'defg'
```

So, we can replace `substr(startIndex, length)` with `substring(startIndex, startIndex + length).`

Reply:
OK. Thanks for looking into this! Was doing the math in my head and got it wrong.